html/captive-portal/lib/captiveportal/PacketFence/Controller/Activate/Sms.pm

```diff
@@ -119,22 +119,30 @@ sub showSmsConfirmation : Private {
 }
 
 sub sms_validation {
-    my ( $self, $c ) = @_;
-    my $logger = Log::Log4perl::get_logger(__PACKAGE__);
+    my ($self, $c) = @_;
+    my $logger = $c->log;
 
     # no form was submitted, assume first time
     my $pin = $c->request->param("pin");
     if ($pin) {
-        $c->log->info("Mobile phone number validation attempt");
-        if ( pf::activation::validate_code($pin) ) {
-            return ( $TRUE, 0 );
-        } else {
-            return ( $FALSE, $GUEST::ERROR_INVALID_PIN );
+        $logger->info("Mobile phone number validation attempt");
+        if ($self->reached_retry_limit($c, 'sms_retries', 3)) {
```

Comment:
s/Should be/Must be/

We must never hardcode things like that in the code, unless if a default and safe value if no configuration is found.